controller/volume_controller.go

```diff
@@ -3008,6 +3095,22 @@ func (vc *VolumeController) updateRecurringJobs(v *longhorn.Volume) (err error)
 	if err = datastore.FixupRecurringJob(v); err != nil {
 		return err
 	}
+
+	// DR volume will not restore volume recurring jobs/groups configuration.
+	if v.Spec.FromBackup != "" && !v.Spec.Standby {
+		if !v.Status.RestoreInitiated {
+			RestoreRecurringJobs, err := vc.ds.GetSettingAsBool(types.SettingNameRestoreVolumeRecurringJobs)
+			if err != nil {
+				vc.logger.WithError(err).Warnf("Failed to get %v setting", types.SettingNameRestoreVolumeRecurringJobs)
+			}
+			if v.Spec.RestoreVolumeRecurringJob == longhorn.RestoreVolumeRecurringJobEnabled || (v.Spec.RestoreVolumeRecurringJob != longhorn.RestoreVolumeRecurringJobDisabled && RestoreRecurringJobs) {
+				if err := vc.restoreVolumeRecurringJobs(v); err != nil {
```

Comment:
One question: For **each** reconcile loop of each restored volume, will the volume controller unmarshal the recurring job label then compare the result with existing recurring jobs? If YES, I would prefer to add the status field back to avoid this kind of repeat & unnecessary checks.

Reply:
No, only once, `v.Status.RestoreInitiated` would become `true` for good after restoration is initialized at `checkAndInitVolumeRestore`.